tests/conditional_ksample/test_cd.py

```diff
@@ -0,0 +1,233 @@
+import numpy as np
+import pandas as pd
+import pytest
+from flaky import flaky
+from sklearn.ensemble import RandomForestClassifier
+
+from pywhy_stats import bregman, kcd
+
+seed = 12345
+rng = np.random.default_rng(seed)
+# number of samples to use in generating test dataset; the lower the faster
+n_samples = 150
+
+
+def _single_env_scm(n_samples=200, offset=0.0, seed=None):
+    # We construct a SCM where X1 -> Y <- X and Y -> Z
+    # so X1 is independent from X, but conditionally dependent
+    # given Y or Z
+    rng = np.random.default_rng(seed)
+
+    X = rng.standard_normal((n_samples, 1)) + offset
+    X1 = rng.standard_normal((n_samples, 1)) + offset
+    Y = X + X1 + 0.05 * rng.standard_normal((n_samples, 1))
+    Z = Y + 0.05 * rng.standard_normal((n_samples, 1))
+
+    # create input for the CD test
+    df = pd.DataFrame(np.hstack((X, X1, Y, Z)), columns=["x", "x1", "y", "z"])
+
+    # assign groups randomly
+    df["group"] = rng.choice([0, 1], size=len(df))
+    return df
+
+
+def _multi_env_scm(n_samples=100, offset=1.5, seed=None):
```

Comment:
To verify my understanding, the difference between the two environments are only the different offsets (0.0 vs 1.5)?

Reply:
Yes so the conditional distribution has shifted in some sense and should "be detected"